docs/www/acls.md

```diff
@@ -0,0 +1,304 @@
+
+- TLS certificates.
+
+If you have your own certificates, either self-signed or issued by a trusted Certificate Authority, you can use them for this guide. Otherwise, feel free to generate them using [this script](https://gist.github.com/0x5d/56422a0c447e58d8ccbfa0ce1fd6bac6).
+
+## Authentication
+
+### Mutual TLS (mTLS)
+
+Mutual TLS, or 2-way TLS, is an authentication method in which the server keeps a set of trusted certificates in the form of a "truststore" file, and all clients attempting to establish a connection must present their certificate.
+
+To enable it, set the `require_client_auth` field to `true` in the required listener's configuration. For example, to enable mTLS for the "external" API listener:
+
+```yaml
+redpanda:
+
+  kafka_api:
+  # The listener declaration. `name` can have any value.
+  - name: internal
+    address: <private IP>
+    port: 9092
+
+  advertised_kafka_api:
+  # The advertised listeners. `name` should match the name of a declared listener.
+  - name: internal
+    address: localhost
+    port: 9092
+
+  kafka_api_tls:
+  # The listener's TLS config. `name` must match the corresponding listener's name.
+  - name: internal
+    enabled: true
+    require_client_auth: true # <- This needs to be enabled!
+    cert_file: <path to PEM-formatted cert file>
+    key_file: <path to PEM-formatted key file>
+    truststore_file: <path to PEM-formatted truststore file>
+```
+
+On `rpk`'s side, you can use the `--tls-key`, `--tls-cert` and `--tls-truststore` flags to have it authenticate and establish a TLS connection:
+
+```cmd
+$ rpk topic create test-topic \
+  --tls-key <path to PEM-formatted key file> \
+  --tls-cert <path to PEM-formatted cert file> \
+  --tls-truststore <path to PEM-formatted truststore file>
+
+Created topic 'test-topic'.
+You may check its config with
+
+rpk topic describe 'test-topic'
+```
+
+> `rpk` defaults to connecting to `localhost:9092`. If you're connecting to a remote broker, you will need to pass `--brokers <node IP>:<kafka API port>`
+
+> For an in-depth guide about enabling TLS and mutual TLS in redpanda check out [our guide on the subject](https://vectorized.io/blog/tls-config/).
+
+### SASL/ SCRAM
+
+Redpanda also supports client authentication via SASL/ SCRAM (that is, the Simple Authentication and Security Layer protocol, using the Salted Challenge Response Authentication Mechanism), which is based on usernames & passwords.
+
+To enable it, set `redpanda.enable_sasl` to `true` in the configuration file, as well as list at least one "superuser" which after created will have permissions for all operations on the clusters.
+
+```yaml
+redpanda
+  
+  enable_sasl: true
+  superusers:
+  - admin
+  # The rest of the config...
+```
+
+Then, set a password for the user by running the following command. Replace `<password>` for a password of your choice. 
+
+```cmd
+$ rpk acl user create \
```

Comment:
Needs `--api-urls`

Note: if the `--api-urls` is not provided, the CLI claims that user was created. We'd need a check there.

Reply:
There's a bug in Sarama that swallows errors :( I added [this check](https://github.com/vectorizedio/redpanda/pull/1311/commits/0484838404c28ef5da96a9d6f7453e98d0e3458e#diff-84989bd84b544cf8e2b3dc5204448c47eed57abd49dc8018f714f1c648db81f8R184) to let the user know it's mandatory. I couldn't do it from cobra directly, though. There's a bug in cobra where marking a Persistent Flag as required will do nothing.